lnrpc/invoicesrpc/addinvoice_test.go

```diff
@@ -0,0 +1,840 @@
+
+	args := h.Mock.Called(chanID)
+	return args.Get(0).(lnwire.ShortChannelID), args.Error(1)
+}
+
+// FetchAllChannels retrieves all open channels currently stored
+// within the database.
+func (h *hopHintsConfigMock) FetchAllChannels() ([]*channeldb.OpenChannel,
+	error) {
+
+	args := h.Mock.Called()
+	return args.Get(0).([]*channeldb.OpenChannel), args.Error(1)
+}
+
+// FetchChannelEdgesByID attempts to lookup the two directed edges for
+// the channel identified by the channel ID.
+func (h *hopHintsConfigMock) FetchChannelEdgesByID(chanID uint64) (
+	*channeldb.ChannelEdgeInfo, *channeldb.ChannelEdgePolicy,
+	*channeldb.ChannelEdgePolicy, error) {
+
+	args := h.Mock.Called(chanID)
+
+	// If our error is non-nil, we expect nil responses otherwise. Our
+	// casts below will fail with nil values, so we check our error and
+	// return early on failure first.
+	err := args.Error(3)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	edgeInfo := args.Get(0).(*channeldb.ChannelEdgeInfo)
+	policy1 := args.Get(1).(*channeldb.ChannelEdgePolicy)
+	policy2 := args.Get(2).(*channeldb.ChannelEdgePolicy)
+
+	return edgeInfo, policy1, policy2, err
+}
+
+// getTestPubKey returns a valid parsed pub key to be used in our tests.
+func getTestPubKey() *secp256k1.PublicKey {
+	pubkeyBytes, _ := hex.DecodeString(
+		"598ec453728e0ffe0ae2f5e174243cf58f2" +
+			"a3f2c83d2457b43036db568b11093",
+	)
+	pubKeyY := new(btcec.FieldVal)
+	_ = pubKeyY.SetByteSlice(pubkeyBytes)
+	pubkey := btcec.NewPublicKey(
+		new(btcec.FieldVal).SetInt(4),
+		pubKeyY,
+	)
+	return pubkey
+}
+
+var populateForcedHintsTestCases = []struct {
+	name          string
+	forcedHints   [][]zpay32.HopHint
+	expectedHints [][]zpay32.HopHint
+	expectedErr   string
+}{{
+	name:          "empty forced hint slice",
+	forcedHints:   [][]zpay32.HopHint{},
+	expectedHints: [][]zpay32.HopHint{},
+}, {
+	name: "populate forced hint slice with two hint paths",
+	forcedHints: [][]zpay32.HopHint{
+		{
+			zpay32.HopHint{ChannelID: 1},
+			zpay32.HopHint{ChannelID: 2},
+		}, {
+			zpay32.HopHint{ChannelID: 3},
+			zpay32.HopHint{ChannelID: 4},
+		},
+	},
+	expectedHints: [][]zpay32.HopHint{
+		{
+			zpay32.HopHint{ChannelID: 1},
+			zpay32.HopHint{ChannelID: 2},
+		}, {
+			zpay32.HopHint{ChannelID: 3},
+			zpay32.HopHint{ChannelID: 4},
+		},
+	},
+}, {
+	name:        "number of hop hint within a route must be positive",
+	forcedHints: [][]zpay32.HopHint{{}},
+	expectedErr: "must be positive",
+}}
+
+func TestPopulateForcedHints(t *testing.T) {
+	for _, tc := range populateForcedHintsTestCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			hopHints, err := populateForcedHints(tc.forcedHints)
+			if tc.expectedErr != "" {
+				require.ErrorContains(t, err, tc.expectedErr)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedHints, hopHints)
+		})
+	}
+}
+
+var shouldIncludeChannelTestCases = []struct {
```

Comment:
The fact that the old tests are deleted (unclear why) and then re-added here makes this difficult to review (also all the test case names are changed as well so that can't be used to spot check), namely at a glance it isn't clear if all the prior tests are replicated here.

Reply:
Will look at the test coverage diff manually on a local env